src/walberla_bridge/src/lattice_boltzmann/LBWalberlaImpl.hpp

```diff
@@ -633,45 +623,37 @@ class LBWalberlaImpl : public LBWalberlaBase {
       (*m_boundary)(&block);
   }
 
-  void integrate_push_scheme() {
+  void integrate_update_velocities_from_pdf(
+      std::shared_ptr<BlockStorage> const &blocks) {
+    for (auto b = blocks->begin(); b != blocks->end(); ++b)
+      (*m_update_velocities_from_pdf)(&*b);
+  }
+
+  void integrate_pull_scheme() {
     auto const &blocks = get_lattice().get_blocks();
     // Reset force fields
     integrate_reset_force(blocks);
-    // LB collide
-    integrate_collide(blocks);
-    m_pdf_streaming_communicator->communicate();
-    // Handle boundaries
-    if (m_has_boundaries) {
-      integrate_boundaries(blocks);
-    }
```

Comment:
Should integrate_boundaires mark pdf and velocity ghost layer communicatoin as pending?

Reply:
I removed it again, as it lead to many tests failing